### ISSUE TITLE
Refactor dataset management components for React/Next.js

### ISSUE THREAD

**Issue by robertbartel** [NOAA-OWP/DMOD#295]

**Comment by robertbartel** [comment]
Blocked by completion of #287.

**Comment by robertbartel** [comment]
There will be a lot of overlap between #294 and #295.

**Comment by robertbartel** [comment]
This is probably superseded by #445.
